core/src/main/scala/kafka/log/Log.scala

```diff
@@ -532,6 +532,16 @@ class Log(@volatile var dir: File,
   }
 
   /**
+   * Close file handlers used by log but don't write to disk. This is used when the disk may have failed
```

Comment:
The line in 709 is there before this patch. That is no longer needed after KIP-112. I kept it there to avoid changing the indention of that code block so that the code diff and the code review can be a bit simpler. I will remove that conversion after most all comments have been addressed.

Besides the motivation of avoid changing code indentation, the current patch only convert IOException to KafkaStorageException if the method needs to handle that IOException before throwing that again. I assume that we don't want to have try/catch around all Java library to convert IOException to KafkaStorageException for code simplicity. As a result our Kafka method may potentially throw both IOException and KafkaStorageException and that is why some code needs to catch both.

Regarding the consistency, are you suggesting that the code that currently catches only the `KafkaStorageException` should be updated to catch both `IOException` and `KafkaStorageException` and treat them as `KafkaStoragException`?